### PR TITLE
Change pprof labels to be prometheus compatible

### LINKED ISSUE #32854: Change pprof labels to be conformat with prometheus spec
### Feature Description

This would be [another port of my changes](https://codeberg.org/forgejo/forgejo/pulls/2933).

There are systems, like pyroscope allowing for continuous profiling which do not allow for `-` in labels leading to upload failure.

To resolve this names like `process-description` can be changed either to `processDescription` or `process_description`.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by TheFox0x7**
Enables scrapping pprof endpoint for continuous profiling

Closes: https://github.com/go-gitea/gitea/issues/32854

---

There's probably room for improvement - for example dropping `graceful_lifecycle` completely and making `with-hammer` into `with: hammer` key value (or something similar). But I'd wait with that for a proper spec in OTel so there there won't be a need to align to spec `n` times.

Name is compliant with both prometheus regex and [OTel recommendation](https://opentelemetry.io/docs/specs/semconv/general/attribute-naming/) (but not to their semantic conventions):
> For each multi-word dot-delimited component of the attribute name separate the words by underscores (i.e. use snake_case).

### CODE REVIEW

**Comment by lunny** [comment]
Is this a breaking change for those upgraded Gitea instance?

**Comment by lunny** [comment]
The pprof labels was introduced in #19202 and I found there is a grafana mixin introduced in #17758. Maybe this can also be added to grafana mixin in the same PR?

File: modules/graceful/manager_common.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ const (
 	watchdogMsg  systemdNotifyMsg = "WATCHDOG=1"
 )
 
+// LifecyclePProfLabel is a label marking manager lifecycle phase
+// Compliant with prometheus key regex
```
Comment (wxiaoguang) [inline modules/graceful/manager_common.go]:
```suggestion
// LifecyclePProfLabel is a label marking manager lifecycle phase
// Making it compliant with prometheus key regex https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
// would enable someone interested to be able to to continuously gather profiles into prometheus.
// Other labels for pprof (in "modules/process" package) should also follow this rule.
```

**Comment by wxiaoguang** [comment]
> Maybe this can also be added to grafana mixin in the same PR?

No, there are different, do not mix them at the moment.

File: modules/graceful/manager_common.go (hunk 1 above)
Comment (TheFox0x7) [reply modules/graceful/manager_common.go]: swapped prometheus with pyroscope as prometheus can't process pprof

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]
Let's measure if it is breaking by whether someone complains :smile:

**Comment by wxiaoguang** [comment]
> Let's measure if it is breaking by whether someone complains 😄

Actually it breaks nothing .....

**Pull request merged**